src/histogram.rs

```diff
@@ -0,0 +1,118 @@
+/// Structure used to create a distribution of node
+/// centrality values for use in a displayed histogram
+
+#[derive(Default, Clone)]
+pub struct Histogram {
+    /// Minimum value of a factor.
+    pub min: f64,
+    /// Maximum value of a factor.
+    pub max: f64,
+    /// store values
+    pub values: Vec<f64>,
+}
+
+impl Histogram {
+    pub fn default() -> Self {
+        Self {
+            min: f64::MAX,
+            max: f64::MIN,
+            values: Vec::new(),
+        }
+    }
+
+    // Store a single value
+    pub fn add(&mut self, value: f64) {
+        self.values.push(value);
+    }
+
+    // Compute the histogram tables, returning that table along
+    // with the value of the maximum count in a single slot.
+    pub fn compute(&mut self, num_counts: usize) -> (Vec<usize>, usize) {
+        for value in self.values.iter() {
+            if *value < self.min {
+                self.min = *value;
+            }
+            if *value > self.max {
+                self.max = *value;
+            }
+        }
+
+        let delta = self.max - self.min;
+        let mut counts = vec![0; num_counts];
+        let mut max_count: usize = 0;
+
+        if delta == 0.0 {
+            return (counts, max_count);
+        }
+
+        for value in self.values.iter() {
+            let part = (*value - self.min) / delta;
+            let mut slot = (part * num_counts as f64).floor() as usize;
+            if slot >= num_counts {
+                slot = num_counts - 1;
```

Comment:
Could you try running such test case?

```
    #[test]
    fn histogram_zero_test() {
        let mut histogram = Histogram {
            ..Histogram::default()
        };

        histogram.add(1.6);
        histogram.add(2.6);
        histogram.add(3.6);
        histogram.add(4.6);
        let (counts, max_count) = histogram.compute(0);
    }
```

Reply:
`saturating_sub()` should be here to prevent `panicked at 'attempt to subtract with overflow'`

Reply:
Your test is added (🙏 ) (and amended), and passes.  The end looks like this:
```
        histogram.add(1.6);
        histogram.add(2.6);
        histogram.add(3.6);
        histogram.add(4.6);
        let (counts, max_count) = histogram.compute(0);
        assert_eq!(counts.len(), 0);
        assert_eq!(max_count, 0);
```

But I don't understand your comments about using `saturating_sub()`.   On what line did you mean?

Reply:
Ah, now I understand (duh).  In any event, if `num_counts` is 0, I exit early.

Reply:
Exactly :) Exiting early was one of the solutions 👍